source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,239 @@
+    uint256 public currentLowerBoundRepPriceInAttoEth; // The lower bound for the price of REP in attoEth
+    uint256 public initialRepSalePrice; // The initial price of REP in attoETH for the current auction
+    uint256 public initialEthSalePrice; // The initial price of ETH in attoREP for the current auction
+    uint256 public lastRepPrice; // The last auction's Rep price in attoETH, regardless of wether the result is used in determining reporting fees
+    uint256 public repPrice; // The Rep price in attoETH that should be used to determine reporting fees during and immediately after an ignored auction.
+    uint256 public currentRepPrice; // The Rep price in attoETH currently being derived via an active auction. Once the auction ends this will be used until it can be officially recorded as the repPrice at the beginning of an ignored auction
+
+    modifier onlyAuthorizedPriceFeeder {
+        require(authorizedPriceFeeders[msg.sender]);
+        _;
+    }
+
+    modifier initializeNewAuctionIfNeeded {
+        if (currentAuctionIndex != getAuctionIndexForCurrentTime()) {
+            initializeNewAuction();
+        }
+        _;
+    }
+
+    function initialize(IUniverse _universe) public beforeInitialized returns (bool) {
+        endInitialization();
+        require(_universe != address(0));
+        universe = _universe;
+        reputationToken = universe.getReputationToken();
+        initializationTime = controller.getTimestamp();
+        authorizedPriceFeeders[controller.owner()] = true;
+        manualRepPriceInAttoEth = Reporting.getAuctionInitialRepprice();
+        lastRepPrice = manualRepPriceInAttoEth;
+        repPrice = manualRepPriceInAttoEth;
+        currentRepPrice = manualRepPriceInAttoEth;
+        bootstrapMode = true;
+        return true;
+    }
+
+    function initializeNewAuction() public returns (bool) {
+        if (currentAuctionType == AuctionType.RECORDED) {
+            repPrice = currentRepPrice;
+        }
+        lastRepPrice = currentRepPrice;
+        currentAuctionType = getAuctionType();
+        uint256 _currentAuctionIndex = getAuctionIndexForCurrentTime();
+        require(currentAuctionType == AuctionType.UNRECORDED || currentAuctionType == AuctionType.RECORDED);
+        require(currentAuctionIndex != _currentAuctionIndex);
+        if (bootstrapped) {
+            bootstrapMode = false;
+        }
+        require(!bootstrapMode || currentAuctionType == AuctionType.UNRECORDED);
+        bootstrapped = true;
+
+        uint256 _auctionRepBalanceTarget = reputationToken.totalSupply() / Reporting.getAuctionTargetSupplyDivisor();
+        uint256 _repBalance = reputationToken.balanceOf(this);
+
+        if (_repBalance < _auctionRepBalanceTarget) {
+            reputationToken.mintForAuction(_auctionRepBalanceTarget.sub(_repBalance));
+        } else {
+            reputationToken.burnForAuction(_repBalance.sub(_auctionRepBalanceTarget));
```

Comment:
We should only burn REP that was sourced from fee payouts.  Excess REP from an attack against the auction should be used to make the following auction larger, thus increasing the cost of a sustained attack against the system with each iteration.

Reply:
Yup, makes sense and will address